ClinGen/clincoded
# Connects to #768. Connects to #763. Population evaluation forms

Instance here: http://768-mc-pop-eval-forms-try2.instance.clinicalgenome.org/

#### Connects to #768:

Added in population tab criteria forms, both disease-independent (BA1 and PM2; mutually exclusive) and disease-dependent (BS1). Removed placeholder text for explanation boxes and increased font size of form header title. Feedback text for saving evaluations reports which criteria evaluations were saved.

*Todo/revisit in future tickets:*

* Add color cues to criteria code labels in forms (relevant to all evaluation forms)
* Better display of criteria explanation (the text currently in blue boxes) - suggestions that came up included mouse-over popups for the explanation text or shifting the blue boxes to the right or left of the checkboxes to save on vertical space. Wait for feedback from curators?? (relevant to all evaluation forms)

#### Connects to #763:

Added in support for disease-specific criteria to evaluation forms functionality.

*Todo/revisit in future tickets:*

* As the ability to connect diseases are not currently fully implemented, the portion of the code testing for the presence of a disease in the interpretation object is *untested*. This needs to be re-visited once the ability to connect a disease to an interpretation is in
* Make disabled checkboxes more evident that they're disabled..?

Screenshot for above tickets:
![image](https://cloud.githubusercontent.com/assets/4326866/16720196/9132d8f6-46e8-11e6-837b-ea71eea6cf4a.png)

*Testing:*

1. Add variant and get to VCI
2. Add interpretation to variant
3. Confirm that the forms for the three criteria code show up in population tab
4. Check that you cannot check the checkboxes for both BA1 and PM2 simultaneously
5. Confirm that you cannot interact with the BS1 fields (no disease means that disease-dependent criteria codes cannot be evaluated upon)
6. Make interpretations and save
7. Confirm that the evaluations are saved correctly and liked under the interpretation object. Also confirm that the external data (`populationData`, in this instance) is correctly saved for the evaluations, and that no evaluation object exists for criteria code BS1 (if a disease is not associated, disease-dependent criteria codes cannot be evaluated upon, and their corresponding evaluation objects should *not* be generated even if they are tied under the same Save button)
